src/set_theory/fixed_points.lean

```diff
@@ -0,0 +1,508 @@
+  f i hi (deriv_bfamily o f a) = deriv_bfamily o f a :=
+by { rw ←family_of_bfamily_enum o f, exact deriv_family_fp (λ _, H _ _) _ _ }
+
+theorem le_iff_deriv_bfamily (H : ∀ i hi, is_normal (f i hi)) {a} :
+  (∀ i hi, f i hi a ≤ a) ↔ ∃ b, deriv_bfamily o f b = a :=
+begin
+  unfold deriv_bfamily,
+  rw ←le_iff_deriv_family,
+  { refine ⟨λ h i, h _ _, λ h i hi, _⟩,
+    rw ←family_of_bfamily_enum o f,
+    apply h },
+  exact λ _, H _ _
+end
+
+theorem fp_iff_deriv_bfamily (H : ∀ i hi, is_normal (f i hi)) {a} :
+  (∀ i hi, f i hi a = a) ↔ ∃ b, deriv_bfamily o f b = a :=
+begin
+  rw ←le_iff_deriv_bfamily H,
+  refine ⟨λ h i hi, le_of_eq (h i hi), λ h i hi, _⟩,
+  rw ←(H i hi).le_iff_eq,
+  exact h i hi
+end
+
+theorem deriv_bfamily_eq_enum_ord (H : ∀ i hi, is_normal (f i hi)) :
+  deriv_bfamily o f = enum_ord (⋂ i hi, function.fixed_points (f i hi)) :=
+begin
+  rw ←eq_enum_ord _ (fp_bfamily_unbounded H),
+  use (deriv_bfamily_is_normal f).strict_mono,
+  rw set.range_eq_iff,
+  refine ⟨λ a, set.mem_Inter₂.2 (deriv_bfamily_fp H a), λ a ha, _⟩,
+  rw set.mem_Inter₂ at ha,
+  rwa ←fp_iff_deriv_bfamily H
+end
+
+end
+
+/-! ### Fixed points of a single function -/
+
+section
+variable {f : ordinal.{u} → ordinal.{u}}
+
+/-- The next fixed point function, the least fixed point of the normal function `f`, at least `a`.
+-/
+def nfp (f : ordinal → ordinal) : ordinal → ordinal :=
+nfp_family (λ _ : unit, f)
+
+theorem nfp_eq_nfp_family (f : ordinal → ordinal) : nfp f = nfp_family (λ _ : unit, f) :=
+rfl
+
+@[simp] theorem sup_iterate_eq_nfp (f : ordinal.{u} → ordinal.{u}) (a) :
+  sup (λ n : ℕ, f^[n] a) = nfp f a :=
+begin
+  refine le_antisymm _ (sup_le.2 (λ l, _)),
+  { rw sup_le,
+    intro n,
+    rw [←list.length_repeat unit.star n, ←nfp_family_iterate_eq_iterate.{0 u} f a],
+    exact le_sup _ _ },
+  { rw nfp_family_iterate_eq_iterate.{0 u} f a l,
+    exact le_sup _ _ },
+end
+
+theorem iterate_le_nfp (f a n) : f^[n] a ≤ nfp f a :=
+by { rw ←sup_iterate_eq_nfp, exact le_sup _ n }
+
+theorem self_le_nfp (f a) : a ≤ nfp f a :=
+iterate_le_nfp f a 0
+
+theorem lt_nfp {a b} : a < nfp f b ↔ ∃ n, a < (f^[n]) b :=
+by { rw ←sup_iterate_eq_nfp, exact lt_sup }
+
+theorem nfp_le {a b} : nfp f a ≤ b ↔ ∀ n, (f^[n]) a ≤ b :=
+by { rw ←sup_iterate_eq_nfp, exact sup_le }
+
+theorem nfp_monotone (hf : monotone f) : monotone (nfp f) :=
+nfp_family_monotone (λ i, hf)
+
+theorem is_normal.apply_lt_nfp {f} (H : is_normal f) {a b} :
+  f b < nfp f a ↔ b < nfp f a :=
+begin
+  unfold nfp,
+  rw ←@apply_lt_nfp_family_iff unit (λ _, f) _ (λ _, H) a b,
+  exact ⟨λ h _, h, λ h, h unit.star⟩
+end
+
+theorem is_normal.nfp_le_apply {f} (H : is_normal f) {a b} : nfp f a ≤ f b ↔ nfp f a ≤ b :=
+le_iff_le_iff_lt_iff_lt.2 H.apply_lt_nfp
+
+theorem is_normal.nfp_le_fp {f} (H : is_normal f) {a b} (ab : a ≤ b) (h : f b ≤ b) : nfp f a ≤ b :=
+nfp_family_le_fp (λ _, H) ab (λ _, h)
+
+theorem is_normal.nfp_fp {f} (H : is_normal f) : ∀ a, f (nfp f a) = nfp f a :=
```

Comment:
Do you want to say that `nfp f a` is in `function.fixed_points f`?

Reply:
Surely that's a clunkier way to reword this statement? They're def-eq anyways.